pkg/aks/check_test.go

```diff
@@ -24,3 +31,83 @@ func Test_generateUniqueLogWorkspace(t *testing.T) {
 		assert.Len(t, got, 63)
 	}
 }
+
+var _ = Describe("checkLogAnalyticsWorkspaceForMonitoring", func() {
+	var (
+		workplacesClientMock *mock_services.MockWorkplacesClientInterface
+		mockController       *gomock.Controller
+	)
+
+	BeforeEach(func() {
+		mockController = gomock.NewController(GinkgoT())
+		workplacesClientMock = mock_services.NewMockWorkplacesClientInterface(mockController)
+	})
+
+	AfterEach(func() {
+		mockController.Finish()
+	})
+
+	It("should return workspaceID if workspace already exists", func() {
+		workspaceName := "workspaceName"
+		workspaceResourceGroup := "resourcegroup"
+		id := "workspaceID"
+		workplacesClientMock.EXPECT().Get(ctx, workspaceResourceGroup, workspaceName).Return(operationalinsights.Workspace{
+			ID: &id,
+		}, nil)
+		workspaceID, err := checkLogAnalyticsWorkspaceForMonitoring(ctx,
+			workplacesClientMock,
+			"eastus", workspaceResourceGroup, "", workspaceName)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(workspaceID).To(Equal(id))
+	})
+
+	It("should create workspace if it doesn't exist", func() {
+		workspaceName := "workspaceName"
+		workspaceResourceGroup := "resourcegroup"
+		id := "workspaceID"
+		workplacesClientMock.EXPECT().Get(ctx, workspaceResourceGroup, workspaceName).Return(operationalinsights.Workspace{}, errors.New("not found"))
+		workplacesClientMock.EXPECT().CreateOrUpdate(ctx, workspaceResourceGroup, workspaceName,
+			operationalinsights.Workspace{
+				Location: to.StringPtr("eastus"),
+				WorkspaceProperties: &operationalinsights.WorkspaceProperties{
+					Sku: &operationalinsights.WorkspaceSku{
+						Name: operationalinsights.WorkspaceSkuNameEnumStandalone,
+					},
+				},
+			},
+		).Return(operationalinsights.WorkspacesCreateOrUpdateFuture{}, nil)
+		workplacesClientMock.EXPECT().AsyncCreateUpdateResult(gomock.Any()).Return(operationalinsights.Workspace{
+			ID: &id,
+		}, nil)
+
+		workspaceID, err := checkLogAnalyticsWorkspaceForMonitoring(ctx,
+			workplacesClientMock,
+			"eastus", workspaceResourceGroup, "", workspaceName)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(workspaceID).To(Equal(id))
+	})
+
+	It("should fail if CreateOrUpdate returns error", func() {
+		workplacesClientMock.EXPECT().Get(ctx, gomock.Any(), gomock.Any()).Return(operationalinsights.Workspace{}, errors.New("not found"))
+		workplacesClientMock.EXPECT().CreateOrUpdate(ctx, gomock.Any(), gomock.Any(), gomock.Any()).Return(operationalinsights.WorkspacesCreateOrUpdateFuture{}, errors.New("error"))
+
+		_, err := checkLogAnalyticsWorkspaceForMonitoring(ctx,
+			workplacesClientMock,
+			"eastus", "workspaceResourceGroup", "", "workspaceName")
+
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("should fail if AsyncCreateUpdateResult returns error", func() {
```

Comment:
Maybe in the future we could add a test where it fails initially but on the next poll it passes, a nice to have for the future.